[BE] 히스토리 삭제 예외 발생 및 테스트 코드 추가

Ojeegu opened the pull request:
## What is this PR? 👓
- 잘못된 히스토리 id를 전달받으면 커스텀 예외가 발생하도록 구현
- 예외에 대한 테스트 코드 추가

## Key changes 🔑
- `HISTORY_NOT_FOUND` 예외 추가
- API 설계에 맞게 `ErrorResponse`를 변경
  - `ErrorDto`를 만들어서 에러 번호, 상태, 메세지를 담도록 구현

## To reviewers 👋
~~- 에러 상태는 API 설계대로 "errorCode"에 담아서 반환했습니다.~~
  ~~- 다만 `errorcode` 패키지의 `ErrorCode` 인터페이스와 헷갈릴까봐 해당 클래스 명은 `ErrorDto`로 설정했습니다.~~
  ~~- `ErrorDto errorCode`의 클래스명을 바꾸거나 API 설계를 변경하는 것도 염두에 두고 있습니다. 의견이 있으시면 말씀해주세요.~~
- `ErrorDto`의 인스턴스 명을 `errorDto`로 변경하고 `@JsonProperty("errorCode")`로 설정하였습니다.
- 확장성을 위해 `CommonError`(스프링 제공 에러)를 넣어두었으나 사용 계획이 없는 경우 삭제하는 것이 좋을 것 같습니다.
  - 현재는 `INVALID_INPUT_FORMAT`만 `Controller`를 구현한 상태입니다.
  - `CommonError`를 계속 쓴다면 추후 `PAGE_NOT_FOUND`도 구현하겠습니다. 

## 테스트 실행 결과
![image](https://github.com/codesquad-team-06/todo-max/assets/107015624/05eb097d-7c01-43bf-a15d-7f2c44e31d72)

## Issues
Closes #7

Ojeegu committed:
Merge branch 'be' of https://github.com/codesquad-team-06/todo-max

Ojeegu committed:
Merge branch 'be' of https://github.com/codesquad-team-06/todo-max

Ojeegu committed:
#7 ✨feat: 커스텀 예외 구현 및 적용

- History ErrorCode 작성
- 유효하지 않는 활동 기록 삭제 시 예외 발생

Ojeegu committed:
#7 ✨feat: 에러 코드 정보를 담은 DTO 추가

Ojeegu committed:
#29 ♻refactor: 에러 발생 시 Response Body 수정

- API 설계와 맞는 Response Body로 수정

Ojeegu committed:
#7 🎨style: 메서드명 및 호출 순서 변경

Ojeegu committed:
#7 ♻refactor: 예외 발생 조건 변경

validate(request)가 false인 경우 예외를 발생시키도록 변경

Ojeegu committed:
#7 ♻refactor: ErrorDto 변수명 변경

API 설계에 맞게 ErrorDto 타입 변수명을 ErrorCode로 변경

Ojeegu committed:
#7 ✅test: 유효하지 않는 id 전달 시 예외 발생하는지 테스트 코드 구현

Ojeegu committed:
#7 🎨style: 줄바꿈 LF로 변경

Ojeegu committed:
#7 🎨style: 코딩 컨벤션 적용

Ojeegu committed:
#7 🎨style: todo 삭제

yonghwankim-dev commented on be/src/main/java/codesquad/todo/history/service/HistoryService.java:
```diff
@@ -29,14 +31,13 @@ public List<HistoryFindAllResponse> getAll() {
 	public boolean deleteByIds(HistoryDeleteRequest request) {
-		if (validate(request)) {
-			throw new RuntimeException("유효하지 않은 히스토리입니다.");
+		if (!validate(request)) {
```
어떤 요청을 검증하는지에 대한 메소드명 변경이 필요할 것 같습니다.

Ojeegu replied:
파라미터로 전달되는 값을 검증하는 기능은 굳이 메소드에 안적어도 된다는 피드백을 이전에 받은 적이 있는데, Id를 추가하는 게 좋을까요?

yonghwankim-dev replied:
그러한 피드백을 받았다면 따로 추가하지 않으셔도 됩니다. 제 주관적인 생각으로는 어떤 것을 검증하는지 메소드명으로 파악이 되지 않아서 그랬습니다.

Ojeegu committed:
#29 🎨style: ErrorDto 인스턴스 명 변경

- errorCode -> errorDto로 변경
- @JsonProperty 어노테이션을 이용해 Json 반환 시 errorCode로 전달하도록 설정
